src/ArduPilotPlugin.cc

```diff
@@ -437,6 +459,29 @@ void ignition::gazebo::systems::ArduPilotPlugin::LoadControlChannels(
       return;
     }
 
+    // set up publisher if relaying the command
+    if (control.type == "COMMAND")
+    {
+      if (controlSDF->HasElement("cmd_topic"))
+      {
+        control.cmdTopic = controlSDF->Get<std::string>("cmd_topic");
+      }
+      else
+      {
+        control.cmdTopic =
```

Comment:
Interesting... makes sense, but it looks like the ThrusterPlugin doesn't honor this. Is this the preferred method?

Reply:
Ignition isn't entirely consistent on this. Some topics are fully scoped along the lines of

```
/world/<world_name>/model/<model_name>/joint/<joint_name>/<topic_type>
```

and others are variously abbreviated. Given an ignition gazebo server can address more than one world I think it future proofs the plugin to default to the full path - but as you note a number of other systems plugins don't follow this pattern.